webapp/packages/core-blocks/src/FormControls/useInputAutocomplete.ts

```diff
@@ -0,0 +1,141 @@
+/*
+ * CloudBeaver - Cloud Database Manager
+ * Copyright (C) 2020-2024 DBeaver Corp and others
+ *
+ * Licensed under the Apache License, Version 2.0.
+ * you may not use this file except in compliance with the License.
+ */
+import { action, computed, observable } from 'mobx';
+import { type RefObject, useEffect, useMemo } from 'react';
+
+import { debounce, isNotNullDefined } from '@cloudbeaver/core-utils';
+
+import { useObservableRef } from '../useObservableRef.js';
+import { type SearchStrategy, useSearch } from '../useSearch.js';
+
+export type InputAutocompleteStrategy = 'startsWith' | 'contains' | 'fuzzy';
+
+interface InputAutocompleteOptions {
+  sourceHints: InputAutocompleteProposal[];
+  matchStrategy?: SearchStrategy;
+  predicate?: (suggestion: InputAutocompleteProposal, lastWord?: string) => boolean;
+}
+
+export interface InputAutocompleteProposal {
+  id: string;
```

Comment:
i suggest to remove this 'id' fields because:
1. it's not in use
2. it's just a "always" random string